### ISSUE TITLE
UNHANDLED default behaviour

### ISSUE THREAD

**Issue by debocsh** [jovotech/jovo-framework#1273]
## I'm submitting a...
- [x] Documentation issue or request
As mentioned in the docs, unhandled will always be prioritized 
<img width="714" alt="image" src="https://user-images.githubusercontent.com/44774270/162606551-04464aef-b904-4b4d-9105-df67ed9c084c.png">
My question is what was the thought behind this ? I would have expected that "Unhandled" will be fulfilled if the input has no matching intent. Here, it seems that **Unhandled** has more weight than other handlers even if the input matched intent in another components. 

I ran into the same scenario that you quote in the docs 
<img width="681" alt="image" src="https://user-images.githubusercontent.com/44774270/162607393-0a880296-53b1-4235-9174-c1643fc602b4.png">

Since it's the default behaviour, should we have also an option to cancel it instead of using _prioritizedOverUnhandled_ or _intentsToSkipUnhandled_ which needs to be added on each new intent ? Does it mean that once I have one component with the Unhandled behaviour, I'll be stuck in this loop ? 
I'll appreciate if you can provide more information/explanation about the UNHANDLED handler role and why this is the **default** behaviour ?

- Jovo Framework version used: 4.1.3 
- Operating System:
     Linux 5.10 Debian
      Node: 16.14.0

**Comment by jrglg** [comment]
You can pass a variable when delegating, for example, that you can use
later from delegated component and override UNHANDLED behaviour.

Remember you can use anotations on handlers ***@***.***)

**Comment by fbublitz** [comment]
If you do not have an UNHANDLED Handler in your Component in question, you achieve the behavior that you want i think. The routing can find Intent handlers in other components. So its basically your decision if you want to create this blocking loop for the user.
If you have not delegated to a component in the previous dialouge step, the routing even searches upwards the hierarchy in the component tree, being able to match a non global handler for the intent in the "mother" component or further up.

**But** if you have delegated to the component and there is no UNHANDLED handler, the component does not resolve, (which makes me wonder what happens on the state stack in that case...) and the routing can only find Global handlers in the Top Level components i think. This is a bit odd, since we explicitly have to declare components that we want to delegate to as child components.

**Comment by jrglg** [comment]
@fbublitz, that's correct. Furthermore it is also possible to have a component twice on the stack. At the moment I have no problems handling these cases, in fact it works quite well if you play with the annotations.

@jankoenig @aswetlow I think it would be really interesting to have **the ability to keep only ONE component on the stack. Avoiding keeping duplicates**, with some mechanism (maybe some annotation). Basically because in more complicated logics, component-level memory doesn't work, since the necessary data may be in another "instance" of the same component.

**Comment by fbublitz** [comment]
@jrglg i checked the behavior again and it turned out i was a partly wrong in my **but** part with the delegation. So when component _X_ delegates to component _Y_ and the next request comes in with intent _myIntent_ , which is not handled in _Y_ . The routing can find _myIntent_ in the _X_ component first (and it does not have to be a global handler) before looking in other components. This is quite nice and more what you would expect.
